apps/coscrad-frontend/src/utils/generic-components/presenters/detail-views/resource-preview-icon.tsx

```diff
@@ -0,0 +1,48 @@
+import { ResourceType } from '@coscrad/api-interfaces';
+import AudiotrackIcon from '@mui/icons-material/Audiotrack';
+import AutoStoriesIcon from '@mui/icons-material/AutoStories';
+import CollectionsBookmarkIcon from '@mui/icons-material/CollectionsBookmark';
+import LanguageIcon from '@mui/icons-material/Language';
+import ListIcon from '@mui/icons-material/List';
+import PhotoIcon from '@mui/icons-material/Photo';
+import PlaceIcon from '@mui/icons-material/Place';
+import SubscriptionsIcon from '@mui/icons-material/Subscriptions';
+import VideocamIcon from '@mui/icons-material/Videocam';
+import VolumeUpIcon from '@mui/icons-material/VolumeUp';
+import { Box } from '@mui/material';
+
+interface ResourcePreviewImageProps {
+    resourceType: ResourceType;
+}
+
+/**
+ *
+ * TODO: move lookupTable into a global location for use by all presenters needing
+ * a generic resource icon
+ *
+ */
+
+const lookupTable: { [K in ResourceType]: JSX.Element } = {
```

Comment:
Ah yes- another case for a single, top-level factory.
```ts
presenterFactory.forResource(ResourceType.photograph).buildIcon()
```
Then to adapt to react native, we just inject a different implementation for the presenters.